### PR TITLE
Blog post, changelog and docs for 1.19

### PR DESCRIPTION
**Pull request by lydell**

### CODE REVIEW

File: changelog_unreleased/TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+6. Change ```jsx to your language. For example, ```yaml.
+
+7. Change the `// Input` and `// Output` comments to the comment syntax of your language. For example, `# Input`.
+
+8. Choose some nice input example code. Paste it along with the output before and after your PR.
+
+-->
+
+#### Title ([#XXXX](https://github.com/prettier/prettier/pull/XXXX) by [@user](https://github.com/user))
```
Comment (sosukesuzuki) [inline changelog_unreleased/TEMPLATE.md]: It seems that the parentheses are unnecessary when attaching the PR and user link. The parentheses do not exist in the [current `CHANGELOG.unreleased.md`](https://raw.githubusercontent.com/prettier/prettier/master/CHANGELOG.unreleased.md) like below:

```md
#### Title ([#XXXX] by [@user])
```

Comment (lydell) [reply changelog_unreleased/TEMPLATE.md]: I don’t understand what you mean?

Comment (sosukesuzuki) [reply changelog_unreleased/TEMPLATE.md]: Sorry...
```suggestion
#### Title ([#XXXX] by [@user])
```

Comment (lydell) [reply changelog_unreleased/TEMPLATE.md]: Then we have to add this at the bottom:

```md
[#XXXX]: https://github.com/prettier/prettier/pull/XXXX
[@user]: https://github.com/user
```

Which means more typing for contributors and more room for error.

Comment (sosukesuzuki) [reply changelog_unreleased/TEMPLATE.md]: I see. I didn't know, sorry.

**Comment by fisker** [comment]
I was think about this the other day, maybe we can do this

use dir name `changelog` 

for released changes we can move `CHANGELOG.md` into this, also we can split this into several pieces files when its became big in future.

for PRs, add subdir `pr` or `pull-request` , peple can put there changelog in  `1234.md` or `1234-1235-1236.md` , we don't need clean them we release a new  version, we will got the full historys.

for unreleased changes , we still use a seperate file `unreleased.md` only index the file name(link to `pr/1234.md`) of there PR, (not sure will this cause conflicts, if so we can use subdir 
 `unreleased` and move them into `pr` when release).

TEMPLATE.md should name `0000.md` or `_TEMPLATE.md` it can be on the top when people open folder

**Comment by lydell** [comment]
@brainkim When running `master` on some code bases I have access to, I got lots of improvements like this:

```diff
       return events;
     }),
     TE.chain(publishEventsToService(dryRun))
-  )().then(E.fold(err => console.error(err), _ => console.log("Went well")));
+  )().then(
+    E.fold(
+      err => console.error(err),
+      _ => console.log("Went well")
+    )
+  );
```

But I also got cases where the output kind of became worse:

```diff
 export const timestampOfFirstCommit = (timelines: Timelines): O.Option<Date> =>
-  pipe(
-    timelines,
-    everyCommitTimestamps,
-    A.sort(ordDate),
-    A.head
-  );
+  pipe(timelines, everyCommitTimestamps, A.sort(ordDate), A.head);

 ) => (msg: unknown) =>
   pipe(
     serviceEventFromMessage(msg),
-    TE.chain(
-      flow(
-        publishServiceEvent(analytics),
-        TE.mapLeft(nackFromError)
-      )
-    )
+    TE.chain(flow(publishServiceEvent(analytics), TE.mapLeft(nackFromError)))
   )().then(messageResponse(logger, msg));

   .catch((err: Error) => {
     logger.error(
-      pipe(
-        O.fromNullable(err.stack),
-        O.getOrElse(constant(err.message))
-      )
+      pipe(O.fromNullable(err.stack), O.getOrElse(constant(err.message)))
     );
     process.exit(1);
   });

     expect(
-      pipe(
-        Changelog.timestampOfFirstCommit([[commit]]),
-        O.toUndefined
-      )
+      pipe(Changelog.timestampOfFirstCommit([[commit]]), O.toUndefined)
     ).toStrictEqual(new Date(0));
   });
```

Thoughts?

**Comment by lydell** [comment]
@fisker I’m not sure if I get your idea completely, but it sounds a little complicated to me. With my proposal, one can do `cat changelog_unreleased/typescript/*.md | pbcopy` and paste that under `### TypeScript` in the blog post you’re writing. Repeat for every category. Move the highlights to the top. Clean some things up. It’s still a little bit manual, but a lot easier than the current situation. And it shouldn’t cause any merge conflicts. I think we need a little bit of human touch to the changelogs for the to be good.

**Comment by lydell** [comment]
@thorn0 cspell is failing on “error recovery parsing”:

```
/home/lydell/forks/prettier/website/blog/2019-11-03-1.19.0.md:196:278 - Unknown word (reco)
CSpell: Files checked: 185, Issues found: 1 in 1 files
```

Do you know how to fix it?

**Comment by thorn0** [comment]
@lydell I fixed it in #6816, see https://github.com/prettier/prettier/pull/6816/files#diff-2a298fb798903102dbeeeae48b89b6d6R396

**Comment by henrikra** [comment]
Everything looks good! I guess this is only one blocking 1.19 release?

**Comment by lydell** [comment]
So, the plan was to do the release yesterday, but since I failed to get access to the npm package that didn’t happen. Unfortunately I won’t have time to do it today, and not tomorrow either. But I could do it on Friday.

If someone else wants to do the release today or tomorrow, that would be great. If so, you need to decide if https://github.com/prettier/prettier/pull/6833 should be part of it or not. (I’d say it should.) If so, here’s a suggestion for a changelog entry:

```diff
diff --git a/website/blog/2019-11-03-1.19.0.md b/website/blog/2019-11-03-1.19.0.md
index 75e05c85..da66a5f4 100644
--- a/website/blog/2019-11-03-1.19.0.md
+++ b/website/blog/2019-11-03-1.19.0.md
@@ -574,6 +574,21 @@ const foo = [abc, def, ghi, jkl, mno, pqr, stu, vwx, yz] as (
 
 ### Flow
 
+#### Add support for enums ([#6833] by [@gkz])
+
+```js
+// Input
+enum E of string {
+  A = "a", B = "b"
+}
+
+// Prettier 1.19
+enum E of string {
+  A = "a",
+  B = "b",
+}
+```
+
 #### Parentheses around arrow functions' return types that have `FunctionTypeAnnotation` nested in `ObjectTypeAnnotation` ([#6717] by [@sosukesuzuki])
 
 This is a workaround for a [bug](https://github.com/facebook/flow/pull/8163) in the Flow parser. Without the parentheses, the parser throws an error.
@@ -1530,6 +1545,7 @@ Thanks to [@fisker] for updating lots of Prettier’s dependencies!
 [#6785]: https://github.com/prettier/prettier/pull/6785
 [#6796]: https://github.com/prettier/prettier/pull/6796
 [#6816]: https://github.com/prettier/prettier/pull/6816
+[#6833]: https://github.com/prettier/prettier/pull/6833
 [@andersk]: https://github.com/andersk
 [@ascorbic]: https://github.com/ascorbic
 [@bakkot]: https://gibhub.com/bakkot
@@ -1542,6 +1558,7 @@ Thanks to [@fisker] for updating lots of Prettier’s dependencies!
 [@fisker]: https://github.com/fisker
 [@g-harel]: https://github.com/g-harel
 [@gavinjoyce]: https://github.com/gavinjoyce
+[@gkz]: https://github.com/gkz
 [@ikatyang]: https://github.com/ikatyang/
 [@jjaffeux]: https://github.com/jjaffeux
 [@jounqin]: https://github.com/JounQin
```

**Comment by evilebottnawi** [comment]
@lydell :disappointed: 

@prettier/core Has anyone already made releases, maybe we can help @lydell ?

**Comment by ekilah** [comment]
I don't suppose anyone knows of a nice way to get this update in the mean time (dying to integrate TS 3.7)?

I ran `yarn add https://github.com/prettier/prettier#{`master`'s last commit hash}` to try, but there were waaaay more dependencies added to my project than previous versions have had, so I assumed there was something wrong with that approach for `prettier`.

**Comment by henrikra** [comment]
Friday is THE day 🥰

**Comment by radiosilence** [comment]
@ekilah `yarn add prettier/prettier` works, or at least adding that in package.json for the version

**Comment by j-f1** [comment]
> I ran `yarn add https://github.com/prettier/prettier#{master's last commit hash}` to try, but there were waaaay more dependencies added to my project than previous versions have had, so I assumed there was something wrong with that approach for `prettier`.

That’s the correct way to do it and also expected. When publishing, we bundle all our dependencies into the generated files so they work in Node and the browser. When you install from GitHub like that, you end up having to pull in all the dependencies separately.

File: website/versioned_docs/version-stable/integrating-with-linters.md [hunk 2]
```diff
@@ -125,7 +125,7 @@ Then in `tslint.json`:
 
 ### Disable formatting rules
 
-[`stylelint-config-prettier`](https://github.com/prettier/stylelint-config-prettier) is a config that disables rules that conflict with Prettier. Add it to your [`devDependencies`], then extend from it within your `.stylelintrc` configuration. Make sure to put it last in the `extends` array, so it gets the chance to override other configs.
+[`stylelint-config-prettier`](https://github.com/prettier/stylelint-config-prettier) is a config that disables rules that conflict with Prettier. Add it to your `devDependencies`, then extend from it within your `.stylelintrc` configuration. Make sure to put it last in the `extends` array, so it gets the chance to override other configs.
```
Comment (fisker) [inline website/versioned_docs/version-stable/integrating-with-linters.md]:
```suggestion
[`stylelint-config-prettier`](https://github.com/prettier/stylelint-config-prettier) is a config that disables rules that conflict with Prettier. Add it to your [`devDependencies`], then extend from it within your `.stylelintrc` configuration. Make sure to put it last in the `extends` array, so it gets the chance to override other configs.
```

Comment (fisker) [reply website/versioned_docs/version-stable/integrating-with-linters.md]: it's from #6682

**Pull request merged**